addons/mrp/models/mrp_workorder.py

```diff
@@ -401,6 +401,36 @@ def _defaults_from_finished_workorder_line(self, reference_lot_lines):
     def record_production(self):
         if not self:
             return True
+        # add missing move for extra component/byproduct
+        for line in self._workorder_line_ids():
```

Comment:
line 119 of the produce wizard, we handle exactly thise usecase: create a move for a owkorder without one

i suggest to remove the implementation in workorder you add here and in the produce wizard already present and add a shared one in abstract workorder